yelp_query.py

```diff
@@ -25,27 +29,38 @@ def query_categories(key, category_alias):
 	# https://www.yelp.com/developers/documentation/v3/category
 	headers = {'Authorization': 'Bearer %s' % key}
 	url ='https://api.yelp.com/v3/categories/{}'.format(category_alias)
-	req=requests.get(url, params=params, headers=headers)
-	
-	#results = json.loads(req.text)
+	req = requests.get(url, params=params, headers=headers)	
 	return req
 
 
-def output_results(req):
-	results = json.loads(req.text)
-	print (results)
-	print ('{} returned points of interest'.format(len(results['businesses'])))
-	print (results['total'])
-	print (results['region'])
+def output_results(result_d):
+	with open('data/result_{}_{}.json'.format(CATEGORY,LOCATION.replace(' ','')), 'w') as out:
+		json.dump(result_d, out)
+	
 
 
 if __name__ == '__main__':
 	key = get_config()
 	# still playing around with the two different endpoints for POI data
 	# results = query_categories(key, 'homeandgarden')
-	req = query_businesses(key, 'Seattle', 'Home Depot','homeandgarden')
-	print('The status code is {}'.format(req.status_code))
-	if req.ok:
-		output_results(req)
-	else:		
-		print(req.raise_for_status())
\ No newline at end of file
+	max_query_results = np.inf
+	offset = 0
+	results_list = []
+	while offset < max_query_results:
```

Comment:
does this ever output? as far as I can tell offset < infinity should continue forever. Maybe a AND statement to catch the end state?

Reply:
yup, it ends (at 52 for this query) the total is returned with each query so it gets reset immediately when returned. although I've got to fix up breaking the loop when there's no return